### ISSUE TITLE
Perpetual pull request recreation.

### ISSUE THREAD

**Issue by Agrael1** [restyled-io/actions#44]
Hello,

I have been using restyle bot for quite a while now, and recently it forced me to use workflows.

However now I'm stuck in a perpetual pull request. Even if I merge PR from restyle it still respawns and fails the pipeline.
https://github.com/Agrael1/Wisdom/pull/182

What could be done?

**Comment by pbrisbin** [comment]
Wow, that's wild. I don't have a 100% clear picture but I at least see something...

When Restyled runs one of the formatters (not sure which), it seems to leave behind `foo.orig` files after fixing a file named `foo`.

The create-pull-request action, by default, commits any and all modifications or additions and includes it in the PR it creates. So, it's seeing these `.orig` files and makes its own commit to add them in the Restyled PR: https://github.com/Agrael1/Wisdom/pull/183/commits/210598aeb08de1bb7e09b5ef968859d287ecd0e8.

These files, by definition, have style issues, so when you merge that PR back into your original PR, Restyled sees them and fixes them. I can see how this could trigger a loop.

However, I would expect see `foo.orig.orig` files in subsequent Restyled PRs, to continue the cycle again, but these subsequent PRs [appear empty](https://github.com/Agrael1/Wisdom/pull/184). So that bit I'm still confused about; I would expect create-pull-request to see no differences between the working directory and the base branch and do nothing. I will have to look into that separately.

I'm open to other ideas, but I'd recommend adding `*.orig` to your project `.gitignore`, to [tell create-pull-request to ignore them](https://github.com/peter-evans/create-pull-request?tab=readme-ov-file#ignore-files)). Given that you do use an auto-formatter that leaves such files behind, that would make sense to me generally. Fixing Restyled would be a bit of a side-effect.

What do you think?

**Comment by Agrael1** [comment]
I have added `.orig` to gitignore, it still goes on forever

**Comment by pbrisbin** [comment]
OK, so it looks like [astyle makes one fix](https://github.com/Agrael1/Wisdom/pull/187/commits/c7d6dfcba405e86c0c736bc46e92f3e3cce645d6), then [clang-format](https://github.com/Agrael1/Wisdom/pull/187/commits/32a35263d2c19ad6ab9ce4003677509e8d050b1b) puts it back. Merging this "fix" just leaves the code as-is was, so the process will of course repeat forever.

The hosted version of Restyled checked for this scenario and didn't open a PR, but the workflow version just uses the `peter-evans/create-pull-request` action, so we are not in control of that decision. I'm surprised `create-pull-request` will create an empty PR, but apparently it does.

If you don't need both `astyle` and `clang-format`, one work-around would be to disable one of them. However, I will think about what I can do within the restyle action to avoid this scenario.

**Comment by pbrisbin** [comment]
OK, I fixed a few fixes that should help here:

- After a restyler runs, don't do `git commit -a`. Use `git commit <path>` with the paths we were restyling.
- After restyler runs, run `clean -d --force` to ensure we don't leave extra files (like `.orig`).
- After restyler runs, check if the commits we made exactly offset each other. If so, undo them.

They're in restyler-v0.5.2.0 which is being built and released now.

Thanks very much for reporting this, and sorry for the bumpy road with the switch to workflow-based.
